src/frequenz/client/base/client.py

```diff
@@ -5,14 +5,102 @@
 
 import abc
 import inspect
+import secrets
+import time
+from base64 import urlsafe_b64encode
 from collections.abc import Awaitable, Callable
 from typing import Any, Generic, Self, TypeVar, overload
 
-from grpc.aio import AioRpcError, Channel
+from cryptography.hazmat.primitives.hashes import SHA256
+from cryptography.hazmat.primitives.hmac import HMAC
+from grpc.aio import (
+    AioRpcError,
+    Channel,
+    ClientCallDetails,
+    Metadata,
+    UnaryUnaryCall,
+    UnaryUnaryClientInterceptor,
+)
 
 from .channel import ChannelOptions, parse_grpc_uri
 from .exception import ApiClientError, ClientNotConnected
 
+RequestT = TypeVar("RequestT")
+ResponseT = TypeVar("ResponseT")
+"""The request and response types of a gRPC method."""
```

Comment:
For the docs to be properly generated, you need to split into one comment per variable:

```suggestion
RequestT = TypeVar("RequestT")
"""The request type of a gRPC method."""

ResponseT = TypeVar("ResponseT")
"""The response type of a gRPC method."""
```

Reply:
Understood, will do.

Reply:
Just use `object` and ditch them! :D

Reply:
ditched those TypeVars